From 55e784435ca552da6d07b623704e6a599a3e413c Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Tue, 24 Jul 2018 19:54:00 -0700
Subject: [PATCH 2048/2328] ASoC: SOF: allow runtime_pm for pci/acpi/spi device

Allow runtime_pm to be enabled after probe is finished for
pci/acpi/spi devices.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/sof-acpi-dev.c | 5 +++++
 sound/soc/sof/sof-pci-dev.c  | 5 +++++
 sound/soc/sof/sof-spi-dev.c  | 5 +++++
 3 files changed, 15 insertions(+)

diff --git a/sound/soc/sof/sof-acpi-dev.c b/sound/soc/sof/sof-acpi-dev.c
index dd268c1..d635e6b 100644
--- a/sound/soc/sof/sof-acpi-dev.c
+++ b/sound/soc/sof/sof-acpi-dev.c
@@ -301,6 +301,11 @@ static int sof_acpi_probe(struct platform_device *pdev)
 			mach->sof_fw_filename);
 	}
 
+	/* allow runtime_pm */
+	pm_runtime_set_autosuspend_delay(dev, SND_SOF_SUSPEND_DELAY);
+	pm_runtime_use_autosuspend(dev);
+	pm_runtime_allow(dev);
+
 	return ret;
 }
 
diff --git a/sound/soc/sof/sof-pci-dev.c b/sound/soc/sof/sof-pci-dev.c
index 7cf55ee..c8bf16f 100644
--- a/sound/soc/sof/sof-pci-dev.c
+++ b/sound/soc/sof/sof-pci-dev.c
@@ -293,6 +293,11 @@ static int sof_pci_probe(struct pci_dev *pci,
 		goto release_regions;
 	}
 
+	/* allow runtime_pm */
+	pm_runtime_set_autosuspend_delay(dev, SND_SOF_SUSPEND_DELAY);
+	pm_runtime_use_autosuspend(dev);
+	pm_runtime_allow(dev);
+
 	return ret;
 
 release_regions:
diff --git a/sound/soc/sof/sof-spi-dev.c b/sound/soc/sof/sof-spi-dev.c
index 57928d1..fb94f24 100644
--- a/sound/soc/sof/sof-spi-dev.c
+++ b/sound/soc/sof/sof-spi-dev.c
@@ -107,6 +107,11 @@ static int sof_spi_probe(struct spi_device *spi)
 	if (ret)
 		platform_device_unregister(sof_pdata->pdev_mach);
 
+	/* allow runtime_pm */
+	pm_runtime_set_autosuspend_delay(dev, SND_SOF_SUSPEND_DELAY);
+	pm_runtime_use_autosuspend(dev);
+	pm_runtime_allow(dev);
+
 	return ret;
 }
 
-- 
2.7.4

